### ISSUE TITLE
Hero's skill aquire algorithm

### ISSUE THREAD

**Issue by crypt47** [ihhub/fheroes2#8587]
### Your question

My personal fun is to beaf up heroes with the skill I like. I often replay the game if lousy skill drops. I would like to know if the order of the skills is pregenerated on some point (seeding) or there is a way to influence the game engine to produce another skill? Thanks.

**Comment by oleg-derevenetz** [comment]
The order of skills depends on the initial map seed (which is generated when the new game is started) and pre-defined for a given hero. See this method for details and comments:

https://github.com/ihhub/fheroes2/blob/4fd43fb5356207581ca65e06440b0c262d8425b3/src/fheroes2/heroes/heroes.cpp#L2170-L2201

**Comment by crypt47** [comment]
Yes, I see. The only way to change upcomming choice is to upgrade hero skills without raising experience. It's a pitty that choice doesn't necessarily include one skill that was already selected before.

**Comment by ihhub** [comment]
Moving this to the discussion section.
